### ISSUE TITLE
Torchat behind socks5 proxy

### ISSUE THREAD

**Issue by geebotron** [prof7bit/TorChat#14]
I am behind a proxy and need to inform torchat to use the proxy to get onto the Internet.

I have tried adding this line to \bin\Tor\torrc.txt:

Socks5Proxy myproxy.example.com:8080

This works with "normal" Tor, but so far I haven't managed to connect with torchat
(At least 15 minutes has elapsed, with no apparent connection)

Any suggestions?

**Comment by prof7bit** [comment]
You did it correctly, this should work.
Are you using the windows version or the Linux version? (from the backslashes I assume its the windows version. Try to uncomment the logging lines in torrc.txt so that tor will write a log file to see whats going on)

**Comment by geebotron** [comment]
Yes, Windows. (I have it inflicted on me ;-) I will test the same problems with Linux when I get chance. I'll let you know what the log says

**Comment by prof7bit** [comment]
On , leatherface  
reply@reply.github.com  
wrote:

> Yes, Windows. (I have it inflicted on me ;-) I will test the same  
> problems with Linux when I get chance. I'll let you know what the log says

The linux version might give you problems here because of a bug in the  
current version that will always overwrite your torrc.txt with the default  
one from /usr/local/lib/torchat/Tor/torrc.txt

Maybe you could also try with the pidgin plugin. here you can create a  
customized torrc.in.txt that will be appended to the aotomatically  
generated torrc. But the most important thing are the tor log files, only  
the log files will give hints about what might be wrong.

**Comment by geebotron** [comment]
Well, firstly I had made the mistake of having another existing tor instance bound to the default ports 9050, 9051, (doh!) but I killed this process and tried again.

This time I ran the version of tor.exe supplied with torchat like this:
\torchat\0.9.9.551\bin\Tor\tor.exe -f \torchat\0.9.9.551\bin\Tor\torrc.txt

These messages hit stdout (last few lines anyway)

Jun 22 14:22:35.808 [info] authority_certs_fetch_missing(): No current certificate known for authority moria1; launching request.
Jun 22 14:22:35.816 [info] authority_certs_fetch_missing(): No current certificate known for authority tor26; launching request.
Jun 22 14:22:35.823 [info] authority_certs_fetch_missing(): No current certificate known for authority dizum; launching request.
Jun 22 14:22:35.831 [info] authority_certs_fetch_missing(): No current certificate known for authority ides; launching request.
Jun 22 14:22:35.839 [info] authority_certs_fetch_missing(): No current certificate known for authority gabelmoo; launching request.
Jun 22 14:22:35.847 [info] authority_certs_fetch_missing(): No current certificate known for authority dannenberg; launching request.
Jun 22 14:22:35.854 [info] authority_certs_fetch_missing(): No current certificate known for authority urras; launching request.
Jun 22 14:22:35.861 [info] authority_certs_fetch_missing(): No current certificate known for authority maatuska; launching request.
Jun 22 14:22:35.869 [warn] Cannot connect to a directory server through a SOCKS proxy!

**Comment by prof7bit** [comment]
hmm... maybe you should ask about this on the tor mailing list because here I'm somewhat out of my depth. I have never tried this myself. Normally I would have said this must work but now when seeing this log message I am not so sure anymore. Maybe the Tor experts can help.
